server/nova-components/CreateFavorite/resources/js/components/form/form.vue

```diff
@@ -0,0 +1,95 @@
+<template>
+
+    <div>
+            <div class="form-container">
+                <p id="form-title">{{title}}</p>
+                <form v-on:submit.prevent="onSubmit" >
+                    
+                    <formInput type="text" name="title"        :value="info.title"     :handleChange="handleChange"  /> 
+                    <formInput type="text" name="author"       :value="info.author"     :handleChange="handleChange" /> 
+                    <formInput type="text" name="description"  :value="info.description"    :handleChange="handleChange" /> 
+                    <formInput type="text" name="urlToImage"   :value="info.urlToImage"    :handleChange="handleChange" /> 
+                    <formInput type="text" name="url"          :value="info.url"    :handleChange="handleChange" /> 
+
+                    <formInput type="date" name="publishedAt" title="Publish Date" :value="info.publishedAt"  :handleChange="handleChange" /> 
+
+                    <selectInput name="user" :options="users" :value="info.user"  :handleChange="handleChange" />
+
+                    <buttonInput type="submit"  text="Create" />
+                </form>
+            </div>
+        </div>
+        
+</template>
+
+
+<script>
+import buttonInput from './buttonInput/buttonInput'
+import formInput from './formInput/formInput'
+import selectInput from './selectInput/selectInput'
```

Comment:
Put all input components in separated single folder of name inputs, Then make index.js folder to import all of them in one line, You can search for **index js as a public interface**